src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Network analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.xaml.cs

```diff
@@ -0,0 +1,269 @@
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+// language governing permissions and limitations under the License.
+
+using Esri.ArcGISRuntime.Data;
+using Esri.ArcGISRuntime.UtilityNetworks;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+
+namespace ArcGISRuntime.WPF.Samples.ConfigureSubnetworkTrace
+{
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        "Configure subnetwork trace",
+        "Network analysis",
+        "Get a server-defined trace configuration for a given tier and modify its traversability scope, add new condition barriers and control what is included in the subnetwork trace result.",
+        "")]
+    [ArcGISRuntime.Samples.Shared.Attributes.OfflineData()]
+    public partial class ConfigureSubnetworkTrace
+    {
+        // Feature service for an electric utility network in Naperville, Illinois.
+        private const string FeatureServiceUrl = "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer";
+        private UtilityNetwork _utilityNetwork;
+
+        // For creating the default starting location.
+        private const string DeviceTableName = "Electric Distribution Device";
+        private const string AssetGroupName = "Circuit Breaker";
+        private const string AssetTypeName = "Three Phase";
+        private const string GlobalId = "{1CAF7740-0BF4-4113-8DB2-654E18800028}";
+
+        // For creating the default trace configuration.
+        private const string DomainNetworkName = "ElectricDistribution";
+        private const string TierName = "Medium Voltage Radial";
+
+        // Utility element to start the trace from.
+        private UtilityElement _startingLocation;
+
+        // Holding the initial conditional expression.
+        private UtilityTraceConditionalExpression _initialExpression;
+
+        // The trace configuration.
+        private UtilityTraceConfiguration _configuration;
+
+        public ConfigureSubnetworkTrace()
+        {
+            InitializeComponent();
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            try
+            {
+                Configuration.IsEnabled = false;
+
+                // Create and load the utility network.
+                _utilityNetwork = await UtilityNetwork.CreateAsync(new Uri(FeatureServiceUrl));
+
+                // Build the choice lists for network attribute comparison.
+                Attributes.ItemsSource = _utilityNetwork.Definition.NetworkAttributes.Where((i => i.IsSystemDefined == false));
```

Comment:
I think this Where clause can be simplified to `i => !i.IsSystemDefined`. Also note that there are extra enclosing parentheses. 

I would recommend trying to make the variable names a little clearer in the lambdas. Not a huge deal, but something like `.Where(netattr => !netAttr.IsSystemDefined)` would make the intent clearer.